PoPPy bugfixes and test coverage

jtniehof opened the pull request:
This PR makes a few improvements to PoPPy:

- The Python backend was completely broken, so added tests using the Python backend instead of C (assuming C is present), as suggested in #310. Fixed Python backend so that these tests pass. This bumps the coverage from 45% to 52%, so still much to do on #310.
- Fixes (and tests) passing lags as a numpy array (Closes #608). There are some slight behavior changes associated. One, it could be argued that the previous checks would have failed for an _empty_ window and lags. However, the error message clearly states "must be populated" not "cannot be empty". So this is an easier check and seems consistent. Second, the error message was previously returned as a string, not printed. PoPPy is fairly inconsistent on print vs. return vs. warnings vs. exceptions. Changing to print seemed to make more sense here, pending an actual overhaul of that interaction.
- A number of style cleanups, particularly comparisons to None

## PR Checklist

- [X] Pull request has descriptive title
- [X] Pull request gives overview of changes
- [X] New code has inline comments where necessary
- [X] (N/A) Any new modules, functions or classes have docstrings consistent with SpacePy style
- [X] (N/A) Added an entry to release notes if fixing a significant bug or providing a new feature
- [X] New features and bug fixes should have unit tests
- [X] Relevant issues are linked to (e.g. `See issue #` or `Closes #`)

jtniehof commented:
I guess the question is whether this merits a release note. The Python backend has been broken for a LONG time so clearly nobody was using it, and it was always kind of a fallback. The lags as array thing seems minor. OTOH, if someone is for some reason stuck with an old SpacePy and wondering why this stuff is broken, the release note might help...

Note also that, because of the way numpy random seeds are structured, the bootstrap will give different results for a given seed between the Python and C backends. I think that's okay (pretty sure we don't guarantee those to be consistent between machines, for instance.)

drsteve reviewed (approved):
Noting that the PR description says "_The Python backend was completely broken_" but doesn't explicitly say that it's now fixed... however, lines 725-728 in the revised version should fix the issue. So I'm approving but would like confirmation and/or a better description in the PR for future reference.

Everything else is good and boy is it rough reading some of this very old code...

jtniehof commented:
> Noting that the PR description says "_The Python backend was completely broken_" but doesn't explicitly say that it's now fixed...

Updated...since tests were added and the CI passes, that seemed implicit, but PEP 20...
